File: LeetCode/src/517.js
var findMinMoves = function(machines) {
    const _sum = sum(machines);
    const [q, d] = [_sum / machines.length, _sum % machines.length];
    
    if (d) return -1;
    
    let result = 0,
        acc = 0;
    machines.forEach(val => {
        const sub = val - q;
        acc += sub;
        result = Math.max(result, Math.abs(acc), sub);
    });
    
    return result;
};

var sum = function(arr) {
    return arr.reduce((a, b) => a + b);
}

File: LeetCode/src/517.py
class Solution:
    def findMinMoves(self, machines: List[int]) -> int:
        _sum = sum(machines)
        target, r = _sum // len(machines), _sum % len(machines)
        if r != 0:
            return -1
        
        _max, total = 0, 0
        for d in machines:
            sub = d - target
            total += sub
            _max = max(_max, sub, abs(total))
            
        return _max
